### PR TITLE
Added drop-down buttons

### PR DESCRIPTION
**Pull request by Cullen-Enerdyne**

### CODE REVIEW

File: imgui.h [hunk 1]
```diff
@@ -490,6 +490,8 @@ namespace ImGui
     // - Most widgets return true when the value has been changed or when pressed/selected
     // - You may also use one of the many IsItemXXX functions (e.g. IsItemActive, IsItemHovered, etc.) to query widget state.
     IMGUI_API bool          Button(const char* label, const ImVec2& size = ImVec2(0, 0));   // button
+    IMGUI_API bool          BeginButtonDropDown(const char* label, ImVec2 buttonSize);
```
Comment (AlexvZyl) [inline imgui.h]: `ImVec2 buttonSize` -> `const ImVec2& size  = ImVec2(0,0)`

Rather keep the API consistent with ImGui.

Comment (Alex-vZyl) [inline imgui.h:493]: `const ImVec2 size  = ImVec2(0,0)`

Rather keep the API consistent with ImGui.

File: imgui_widgets.cpp [hunk 2]
```diff
@@ -720,6 +720,61 @@ bool ImGui::Button(const char* label, const ImVec2& size_arg)
...
+
+    float x = ImGui::GetCursorPosX();
+    float y = ImGui::GetCursorPosY();
+
+    ImVec2 size(20, buttonSize.y);
+    bool pressed = ImGui::Button("##", size);
+
+    // Arrow
+    ImVec2 center(window->Pos.x + x + 10, window->Pos.y + y + buttonSize.y / 2);
+    float r = 8.f;
```
Comment (AlexvZyl) [inline imgui_widgets.cpp:739]: I think `r` should scale based on the button size.

Comment (Alex-vZyl) [inline imgui_widgets.cpp:739]: I think `r` should scale based on the button size.

File: imgui_widgets.cpp [hunk 3]
```diff
@@ -720,6 +720,61 @@ bool ImGui::Button(const char* label, const ImVec2& size_arg)
     return ButtonEx(label, size_arg, ImGuiButtonFlags_None);
 }
 
+bool ImGui::BeginButtonDropDown(const char* label, ImVec2 buttonSize)
+{
+    ImGui::SameLine(0.f, 0.f);
```
Comment (Alex-vZyl) [inline imgui_widgets.cpp:725]: I don't think this should be here, calling `SameLine` is the user's  responsibility.

Comment (AlexvZyl) [inline imgui_widgets.cpp:725]: I don't think this should be here, calling `SameLine` is the user's  responsibility.  What happens if you don't want it to be on the same line?

File: imgui_widgets.cpp [hunk 4]
```diff
@@ -720,6 +720,61 @@ bool ImGui::Button(const char* label, const ImVec2& size_arg)
...
+
+    // Arrow
+    ImVec2 center(window->Pos.x + x + 10, window->Pos.y + y + buttonSize.y / 2);
+    float r = 8.f;
+    center.y -= r * 0.25f;
+    ImVec2 a = center + ImVec2(0, 1) * r;
+    ImVec2 b = center + ImVec2(-0.866f, -0.5f) * r;
+    ImVec2 c = center + ImVec2(0.866f, -0.5f) * r;
+
+    window->DrawList->AddTriangleFilled(a, b, c, GetColorU32(ImGuiCol_Text));
```
Comment (AlexvZyl) [inline imgui_widgets.cpp:745]: Nice.  One of the reasons ImGui is so cool.

Comment (Alex-vZyl) [inline imgui_widgets.cpp:745]: Nice.

**Review by AlexvZyl — CHANGES_REQUESTED** [review]

File: imgui_widgets.cpp (hunk 2 above)
Comment (AlexvZyl) [reply imgui_widgets.cpp:739]: Maybe a,b and c should also change based on the shape?

**Comment by AlexvZyl** [comment]
@Cullen-Enerdyne Changed it so that we merge into `docking`.  I would rather not merge `master` into `docking` and let ocornut handle that.

File: imgui_widgets.cpp (hunk 3 above)
Comment (Cullen-Enerdyne) [reply imgui_widgets.cpp:725]: This doesn't make sense given how drop-downs work. The drop-down isn't really appropriate on the next line. It is inherently a horizontal feature.

File: imgui_widgets.cpp (hunk 2 above)
Comment (Cullen-Enerdyne) [reply imgui_widgets.cpp:739]: You're welcome to make that change if you can work out an appropriate value. 

b and c are colour variables. Why should they be influenced by shape?

File: imgui.h (hunk 1 above)
Comment (Cullen-Enerdyne) [reply imgui.h]: Happy with this.

**Comment by Cullen-Enerdyne** [comment]
Isn't master on our fork? I don;t see what this has to do with ocornut.

File: imgui_widgets.cpp (hunk 2 above)
Comment (AlexvZyl) [reply imgui_widgets.cpp:739]: Woops, misread it then.

File: imgui_widgets.cpp (hunk 3 above)
Comment (Alex-vZyl) [reply imgui_widgets.cpp:725]: So I can't put a dropdown button below something if I want to?

Comment (Alex-vZyl) [reply imgui_widgets.cpp:725]: image
We can't do ths.

**Pull request merged**